src/main/java/com/teamacronymcoders/base/materialsystem/items/ItemMaterialPart.java

```diff
@@ -45,7 +45,12 @@ public ItemMaterialPart(MaterialUser materialUser) {
 
     @Override
     public int getColorFromItemstack(@Nonnull ItemStack itemStack, int tintIndex) {
-        return tintIndex == 0 ? this.getMaterialPartFromItemStack(itemStack).getColor() : -1;
+        MaterialPart materialPart = this.getMaterialPartFromItemStack(itemStack);
+        if (materialPart.getPart().hasOverlayTexture()) {
```

Comment:
Really makes sense still to just keep tintIndex the same for everything? Unless layers do equal the tintIndex in items?

Reply:
Tintindex is the layer from my understanding

Reply:
Fair enough, as long as it works.